src/test/java/flight.reservation/ScenarioTest.java

```diff
@@ -199,8 +203,8 @@ public void initCreditCard() {
             @DisplayName("then the booking should succeed")
             void thenTheBookingShouldSucceed() throws NoSuchFieldException {
                 ScheduledFlight scheduledFlight = schedule.searchScheduledFlight(flight.getNumber());
-                FlightOrder order = customer.createOrder(Arrays.asList("Max"), Arrays.asList(scheduledFlight), 100);
-                boolean processed = order.processOrderWithCreditCard(creditCard);
+                FlightOrder order = customer.createOrder(Collections.singletonList("Max"), Collections.singletonList(scheduledFlight), 100);
+                boolean processed = creditCard.processOrder(order);
```

Comment:
apparently you have changed the existing API of `FlightOrder` (might be a problem for other dependent code)